### PR TITLE
Look up S3 scratch bucket URL dynamically based on AWS user name

### PR DESCRIPTION
**Pull request by vinceatbluelabs**

### CODE REVIEW

File: records_mover/session.py [hunk 1]
```diff
@@ -53,7 +59,25 @@ def _infer_scratch_s3_url(session_type: str) -> Optional[str]:
...
+            #
+            # Automatically add IAM username after specified directory URL
+            #
+            root_home_directories = cfg['aws'].get('root_home_directories', False)
+            if root_home_directories:
+                #
+                # TODO: Which creds should be used here?  Should this
+                # determination be held until we need to determine the
+                # bucket, and the creds be passed in?  So this is a
+                # function rather than a value?https://bluelabs.slack.com/archives/G1MD1GKJB/p1590149305003500
```
Comment (bluelabsbutler) [inline records_mover/session.py:71]: E501 line too long (110 > 100 characters)

File: records_mover/session.py [hunk 2]
```diff
@@ -53,7 +59,25 @@ def _infer_scratch_s3_url(session_type: str) -> Optional[str]:
...
+                #
+                aws_iam_user_name = self._infer_aws_iam_user_name()
```
Comment (bluelabsbutler) [inline records_mover/session.py:73]: F821 undefined name 'self'

File: records_mover/session.py [hunk 3]
```diff
@@ -53,7 +59,25 @@ def _infer_scratch_s3_url(session_type: str) -> Optional[str]:
...
         s3_scratch_url: Optional[str] = cfg['aws'].get('s3_scratch_url')
         if s3_scratch_url is not None:
-            return s3_scratch_url
+            #
+            # Automatically add IAM username after specified directory URL
+            #
+            root_home_directories = cfg['aws'].get('root_home_directories', False)
+            if root_home_directories:
+                #
+                # TODO: Which creds should be used here?  Should this
```
Comment (bluelabsbutler) [inline records_mover/session.py:68]: Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker

**Comment by vinceatbluelabs** [comment]
Superceded by https://github.com/bluelabsio/records-mover/pull/69

**Pull request closed without merging**